lib/socket.js

```diff
@@ -62,20 +61,64 @@ class Transport extends EventEmitter {
   //
   _onSocketData(chunk) {
     const packets = [];
-    this._buffer += chunk;
+    let parsedLength;
+    let newPartLength;
+
+    if (this._buffer) {
+      newPartLength = chunk.indexOf(0);
+      if (newPartLength === -1) {
+        this._buffer = Buffer.concat(
+          [this._buffer, chunk],
+          this._buffer.length + chunk.length
+        );
+        this._checkBufferSize();
+        return;
+      }
+      if (!this._parseRemains(chunk.slice(0, newPartLength), newPartLength)) {
+        return;
+      }
+      chunk = chunk.slice(newPartLength + 1);
+      this._buffer = null;
+    }
 
     try {
-      this._buffer = jsrs.parseNetworkPackets(this._buffer, packets);
+      parsedLength = jsrs.parseNetworkPackets(chunk, packets);
     } catch (error) {
       this.socket.destroy(error);
       return;
     }
 
+    if (chunk.length !== parsedLength) {
+      this._buffer = chunk.slice(parsedLength);
+    }
+
     const packetsCount = packets.length;
     for (let i = 0; i < packetsCount; i++) {
       this.emit('packet', packets[i]);
     }
 
+    if (this._buffer) {
+      this._checkBufferSize();
+    }
+  }
+
+  _parseRemains(newPart, newPartLength) {
```

Comment:
I do not understand the reason for "Remains", as I see this function actually parses the main part and what remains is the latter part of `chunk` after slice.